src/electionguard/group.cpp

```diff
@@ -595,24 +596,34 @@ namespace electionguard
 
     unique_ptr<ElementModP> pow_mod_p(const ElementModP &base, const ElementModP &exponent)
     {
-        const auto &p = P();
-
         // HACL's input constraints require the exponent to be greater than zero
         if (const_cast<ElementModP &>(exponent) == ZERO_MOD_P()) {
             return ElementModP::fromUint64(1UL);
         }
 
         uint64_t result[MAX_P_LEN] = {};
-        bool success = Bignum4096::modExp(p.get(), base.get(), MAX_P_SIZE, exponent.get(),
-                                          static_cast<uint64_t *>(result));
-        if (!success) {
-            throw runtime_error("pow_mod_p: mod operation failed");
-        }
+        CONTEXT_P().modExp(base.get(), MAX_P_SIZE, exponent.get(), static_cast<uint64_t *>(result));
         return make_unique<ElementModP>(result, true);
     }
 
     unique_ptr<ElementModP> pow_mod_p(const ElementModP &base, const ElementModQ &exponent)
     {
+        // HACL's input constraints require the exponent to be greater than zero
+        if (const_cast<ElementModQ &>(exponent) == ZERO_MOD_Q()) {
+            return ElementModP::fromUint64(1UL);
+        }
+
+        // check if we have a lookup table initialized for this element
+        if (base.isFixedBase()) {
+            // TODO: use a smaller key
```

Comment:
calculating this key and storing it is definitely not as performant as it could be but it is sufficient for proving out the model. i could be convinced it shoudl be fixed now, but at the very least should be another ticket.